src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1309,7 +1326,7 @@ private ArrayTypeSymbol VisitArrayInitializer(BoundArrayCreation node)
                 }
                 else
                 {
-                    elementType = TypeSymbolWithAnnotations.Create(bestType, isNullableIfReferenceType: BestTypeInferrer.GetIsNullable(resultBuilder));
+                    elementType = TypeSymbolWithAnnotations.Create(bestType, BestTypeInferrer.GetNullableAnnotation(bestType, resultBuilder));
```

Comment:
> Since the only use of bestType inside BestTypeInferrer.GetNullableAnnotation is to check its IsPossiblyNullableReferenceTypeTypeParameter, could we just pass that bool value?

We could, but I wanted to avoid exposing all consumers to this concept and make them worry about where to get this value from and what is the right way to get it.

---
In reply to: [232029655](https://github.com/dotnet/roslyn/pull/30913#discussion_r232029655) [](ancestors = 232029655)